File: src/main/java/Epoch.java
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.interactions.Actions;

/**
 * Created by eugenel on 11/2/16.
 */
public class Epoch {

    WebDriver driver;

    public Epoch(WebDriver driver){
        this.driver=driver;
    }

    public String getEpochData() {
        WebElement clockField = driver.findElement(By.id("ecclock"));
        return clockField.getText();
    }

    public String getGMT(){
        WebElement fieldGMT = driver.findElement(By.id("result1"));
        return fieldGMT.getText();
    }

    public void clickTimeStamp(){
        WebElement bTimestamp = driver.findElement(By.xpath("//*[@id=\"ef\"]/button[1]"));
        bTimestamp.click();
    }

    public String getTimeZone(){
        WebElement fieldTimeZone = driver.findElement(By.xpath("//*[@id=\"result1\"]/a"));
        return fieldTimeZone.getText();
    }

    //Get current date 01/01/2016 00:00
    public String getDate(){

        WebElement fieldYear = driver.findElement(By.id("hcinput"));
        WebElement fieldMonth = driver.findElement(By.id("mm"));
        WebElement fieldDay = driver.findElement(By.name("dd"));
        WebElement fieldHour = driver.findElement(By.name("hh"));
        WebElement fieldMin = driver.findElement(By.name("mn"));
        WebElement fieldSec = driver.findElement(By.name("ss"));

        return " Year: "+fieldYear.getAttribute("value")+
                " Month: "+fieldMonth.getAttribute("value")+
                " Day: "+fieldDay.getAttribute("value")+
                " Hour: "+fieldHour.getAttribute("value")+
                " Minutes: "+fieldMin.getAttribute("value")+
                " Seconds: "+fieldSec.getAttribute("value");
    }

    //Clear all the data from the page
    public void clear(){
        WebElement link = driver.findElement(By.xpath("//*[@id=\"contentcolumn\"]/div[2]/p[3]/a"));
        link.click();
    }

    //Set 2016 year beginning
    public void setDate(){

        WebElement fieldYear = driver.findElement(By.id("hcinput"));
        WebElement fieldMonth = driver.findElement(By.id("mm"));
        WebElement fieldDay = driver.findElement(By.name("dd"));
        WebElement fieldHour = driver.findElement(By.name("hh"));
        WebElement fieldMin = driver.findElement(By.name("mn"));
        WebElement fieldSec = driver.findElement(By.name("ss"));

        fieldYear.sendKeys("2016");
        fieldMonth.sendKeys("01");
        fieldDay.sendKeys("01");
        fieldHour.sendKeys("01");
        fieldMonth.sendKeys("00");
        fieldMin.sendKeys("00");
        fieldSec.sendKeys("00");

    }

    //Receive timestamp of the 2016 beginning
    public String getTimeStamp(){
        WebElement bToTimestamp = driver.findElement(By.xpath("//*[@id=\"hf\"]/table[3]/tbody/tr/td/button"));
        WebElement Timestamp2016 = driver.findElement(By.id("result2"));
        bToTimestamp.click();
        return Timestamp2016.getText();
    }


}

File: src/Epoch.js
/**
 * Created by eugenel on 3/2/17.
 */
var webdriver = require('selenium-webdriver');

var By = require('selenium-webdriver').By;
var until = require('selenium-webdriver').until;

var driver = new webdriver.Builder().forBrowser('firefox').build();

driver.get("http://www.epochconverter.com/");

driver.manage().window().maximize();

getEpochData();
getGMT("GMT still not shown");
clickTimeStamp();
getGMT("Current: ");
getTimeZone();
getDate();
clear();
setDate();
getTimeStamp();

driver.quit();


///Functions
function getEpochData() {
    getData("ecclock", "Epoch date: ");
}

function getGMT(state) {
    getData("result1", state);
}

function clickTimeStamp() {
    epochClick("//*[@id=\"ef\"]/button[1]");
}

function getTimeZone(){
    getDatabyPath("//*[@id=\"result1\"]/a", "Your time Zone: ");
}

function getDate(){
    getAttribute("hcinput", "Year: ");
    getAttribute("mm", " Month: ");
    getAttributebyName("dd", " Day: ");
    getAttributebyName("hh", " Hour: ");
    getAttributebyName("mn", " Munites: ");
    getAttributebyName("ss", " Seconds: ");
}

function setDate(){
    setKeys("hcinput", "2017");
    setKeys("mm", "01");
    setKeysbyName("dd", "01");
    setKeysbyName("hh", "00");
    setKeysbyName("mn", "00");
    setKeysbyName("ss", "00");
}


function clear(){
    epochClick("//*[@id=\"contentcolumn\"]/div[2]/p[3]/a")
}

function getTimeStamp(){
    epochClick("//*[@id=\"hf\"]/table[3]/tbody/tr/td/button");
    getData("result2", "After update: " );
}

//Helper functions

function getData(ID, printText) {
    driver.wait(until.elementLocated(By.id(ID)), 10000, "Failed to locate").then( function(){
            driver.findElement(By.id(ID)).getText().then (
                function(txt){
                    console.log(printText+txt);
                }
            )
        }
    );
}

function getAttribute(ID, printText) {
    driver.wait(until.elementLocated(By.id(ID)), 10000, "Failed to locate").then( function(){
            driver.findElement(By.id(ID)).getAttribute("value").then (
                function(txt){
                    console.log(printText+txt);
                }
            )
        }
    );
}

function getAttributebyName(ID, printText) {
    driver.wait(until.elementLocated(By.name(ID)), 10000, "Failed to locate").then( function(){
            driver.findElement(By.name(ID)).getAttribute("value").then (
                function(txt){
                    console.log(printText+txt);
                }
            )
        }
    );
}

function getDatabyPath(ID, printText) {
    driver.wait(until.elementLocated(By.xpath(ID)), 10000, "Failed to locate").then( function(){
            driver.findElement(By.xpath(ID)).getText().then (
                function(txt){
                    console.log(printText+txt);
                }
            )
        }
    );
}

function epochClick(ID) {
    driver.wait(until.elementLocated(By.xpath(ID)), 10000, "Failed to locate").then( function(){
            driver.findElement(By.xpath(ID)).click()
        }
    );
}

function setKeys(ID, keys){
    driver.wait(until.elementLocated(By.id(ID)), 10000, "Failed to locate").then( function(){
            driver.findElement(By.id(ID)).sendKeys(keys)
        }
    );
}

function setKeysbyName(ID, keys){
    driver.wait(until.elementLocated(By.name(ID)), 10000, "Failed to locate").then( function(){
            driver.findElement(By.name(ID)).sendKeys(keys)
        }
    );
}